### PR TITLE
[OZ#2: L-06] The sharePrice function should not revert

### PR DESCRIPTION
**Pull request by Robsonsjre**
The `sharePrice` function in the STETHVault contract calculates the amount of stETH needed in order to mint one vault share. However, the calculation does not take into account that `totalSupply` can return zero when no deposits have been made to the vault and therefore no vault shares have been minted. Consequently, when the `totalSupply` function returns zero the `mulDiv` operation will revert.

Consider calling `convertToAssets(10**sharePriceDecimals)` in the `sharePrice` function instead of using the manual calculation as `convertToAssets` handles the zero shares edge case.

### CODE REVIEW

File: test/vaults/STETHVault.ts [hunk 1]
```diff
@@ -198,6 +198,14 @@ describe('STETHVault', () => {
...
 
   describe('Reading functions', () => {
+    it('sharePrice should not revert in the case of 0 totalSupply', async () => {
+      const sharePriceDecimals = await vault.sharePriceDecimals()
+      const totalSupply = await vault.totalSupply()
+      expect(totalSupply).to.be.eq(0)
+      const initialSharePrice = await vault.sharePrice()
+      expect(initialSharePrice).to.be.eq(BigNumber.from('10').pow(sharePriceDecimals))
+      
+    })
```
Comment (ggviana) [inline test/vaults/STETHVault.ts]:
```suggestion
    })
    
```

**Review by ggviana — APPROVED** [review]
